packages/gitbook/src/components/hooks/useHasBeenInViewport.ts

```diff
@@ -0,0 +1,38 @@
+'use client';
+import { useEffect, useState } from 'react';
+
+const HAS_INTERSECTION_OBSERVER = typeof IntersectionObserver !== 'undefined';
+
+/**
+ * Return true if the element has been in the viewport once.
+ */
+export function useHasBeenInViewport(
+    containerRef: React.RefObject<HTMLElement>,
+    options?: Pick<IntersectionObserverInit, 'root' | 'rootMargin' | 'threshold'>,
+) {
+    const [visible, setVisible] = useState(!HAS_INTERSECTION_OBSERVER);
```

Comment:
With SSR, it's going to be true on the server, and then false on the client 🤔 

I think you need `false` here and then set it to true in a hook.

Reply:
SSR ❤️